packages/teams-js/src/private/externalAppAuthenticationForCEA.ts

```diff
@@ -0,0 +1,269 @@
+   * @beta
+   * @hidden
+   * Signals to the host to perform SSO authentication for the application specified by the app ID, and then send the authResult to the application backend.
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId Id of the application backend for which the host should attempt SSO authentication.
+   * @param conversationId ConversationId To tell the bot what conversation the calls are coming from
+   * @param authTokenRequest Parameters for SSO authentication
+   * @throws InvokeError if the host encounters an error while authenticating
+   * @returns A promise that resolves when authentication succeeds and rejects with InvokeError on failure
+   */
+  export async function authenticateWithSSO(
+    appId: AppId,
+    conversationId: string,
+    authTokenRequest: externalAppAuthentication.AuthTokenRequestParameters,
+  ): Promise<void> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    validateId(conversationId, new Error('conversation id is not valid.'));
+
+    const [error] = await sendMessageToParentAsync<[externalAppAuthentication.InvokeError]>(
+      getApiVersionTag(
+        externalAppAuthenticationTelemetryVersionNumber,
+        ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithSSO,
+      ),
+      ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithSSO,
+      [appId.toString(), conversationId, authTokenRequest.claims, authTokenRequest.silent],
+    );
+    if (error) {
+      throw error;
+    }
+  }
+
+  /**
+   * @beta
+   * @hidden
+   * Signals to the host to perform authentication using the given authentication parameters and then send the auth result to the application backend.
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId ID of the application backend to which the request and authentication response should be sent. This must be a UUID
+   * @param conversationId ConversationId To tell the bot what conversation the calls are coming from
+   * @param authenticateParameters Parameters for the authentication pop-up
+   * @throws InvokeError if the host encounters an error while authenticating
+   * @returns A promise that resolves from the application backend and rejects with InvokeError if the host encounters an error while authenticating
+   */
+  export async function authenticateWithOauth(
+    appId: AppId,
+    conversationId: string,
+    authenticateParameters: externalAppAuthentication.AuthenticatePopUpParameters,
+  ): Promise<void> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    validateId(conversationId, new Error('conversation id is not valid.'));
+
+    // Ask the parent window to open an authentication window with the parameters provided by the caller.
+    const [error] = await sendMessageToParentAsync<[externalAppAuthentication.InvokeError]>(
```

Comment:
Would be great to consider using `sendAndUnwrap` so that we can directly get error instead of `[error]`